spring-cloud-data-rest/src/main/java/org/springframework/cloud/data/rest/controller/StreamController.java

```diff
@@ -211,6 +221,38 @@ private void undeployStream(StreamDefinition stream) {
 		}
 	}
 
+	private String calculateStreamState(String name) {
+		List<ModuleStatus> moduleStates = new ArrayList<ModuleStatus>();
+		StreamDefinition stream = repository.findOne(name);
+		for (ModuleDefinition module : stream.getModuleDefinitions()) {
+			moduleStates.add(deployer.status(ModuleDeploymentId.fromModuleDefinition(module)));
+		}
+
+		Set<ModuleStatus.State> states = new HashSet<>();
+		for (ModuleStatus status : moduleStates) {
+			states.add(status.getState());
+		}
+
+		logger.debug("states: {}", states);
+
+		// todo: this requires more thought...
+		if (states.contains(ModuleStatus.State.failed)) {
+			return ModuleStatus.State.failed.toString();
+		}
+		else if (states.contains(ModuleStatus.State.incomplete)) {
+			return ModuleStatus.State.incomplete.toString();
+		}
+		else if (states.contains(ModuleStatus.State.deploying)) {
+			return ModuleStatus.State.deploying.toString();
+		}
+		else if (states.contains(ModuleStatus.State.deployed)) {
```

Comment:
If one of the states is `unknown` but the one other has `deployed` then we would return `deployed` here. May be we can make sure all the states are `deployed` when returning `deployed`?